internal/controllers/reconciliation/integration_test.go

```diff
@@ -893,8 +903,7 @@ func TestHelmOwnershipTransfer(t *testing.T) {
 
 	// Wait for the composition to be sync'd - it shouldn't delete the resource
 	testutil.Eventually(t, func() bool {
-		upstream.Get(ctx, client.ObjectKeyFromObject(comp), comp)
-		return comp.Status.CurrentSynthesis != nil && comp.Status.CurrentSynthesis.ObservedCompositionGeneration == comp.Generation && comp.Status.CurrentSynthesis.Reconciled != nil
+		return errors.IsNotFound(upstream.Get(ctx, client.ObjectKeyFromObject(comp), comp))
```

Comment:
the logic was just... wrong before. It was possible for the composition to be deleted before the check